kube-deploy/roles/kube-addons/tasks/addons-dashboard.yml

```diff
@@ -14,7 +14,18 @@
 # limitations under the License.
 #
 
+#For ARM support only
+- name: Checking for ARM support
+  shell: uname -a | grep -i "arm"
+  register: arm_enabled
+  ignore_errors: true
+
 - name: deploying the kubernetes dashboard
   shell: kubectl create -f https://rawgit.com/kubernetes/dashboard/master/src/deploy/kubernetes-dashboard.yaml
-  when: kube_addons_enabled and "{{kube_addons.kube_dashboard is defined}}"
+  when: kube_addons_enabled and "{{kube_addons.kube_dashboard is defined}}" and arm_enabled.rc != 0
+  ignore_errors: true
+
+- name: deploying the kubernetes dashboard - ARM support only
+  shell: curl -sSL https://rawgit.com/kubernetes/dashboard/master/src/deploy/kubernetes-dashboard.yaml | sed "s/amd64/arm64/g" | kubectl create -f -
```

Comment:
Regarding your first question, that logic is not quite correct.  There are two common ARM architectures that we should support, `arm` and `arm64`.  They are fairly popular, think of Raspberry Pi's and their competitors with more modern architectures (Odroid C2, Pine64).  So we need to conditionally check if the device is `arm64` or `arm` and then do a sed replacement based on that.  The tricky part is that `uname -m` doesn't return `arm` or `arm64` directly.  It returns either `arm<>` or `aarch64`.   The case statement from my [earlier link](https://github.com/kubernetes/kubernetes/blob/85d328f8da12f0f0e5b7e6dd2356e56cd584fc3e/hack/local-up-cluster.sh#L234) gives an example of how to translate `uname -m` to the right image name.  Order of checks do matter.

> I am trying to think through this from the perspective of supporting not only ARM and x86 but others as well. Do you know if any other architectures are supported in Flannel, or is it just x86/64 and ARM?

My goal is the same regarding support for other architectures, not just arm boards.  Flannel currently has released images for amd64, arm, arm64, and ppc64le.  I came up with that list from [kubernetes container registry](https://console.cloud.google.com/kubernetes/images/list?location=GLOBAL&project=google-containers).